pkg/kubernetes/nodestore_test.go

```diff
@@ -0,0 +1,294 @@
+				},
+			},
+			wantNodes: map[string]map[string]string{
+				"node-1": {
+					"some-label":  "val",
+					"other-label": "val",
+					"new-label":   "new-val",
+				},
+			},
+		},
+		{
+			name: "Update node labels with new label values for existing node",
+			node: v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "node-1",
+					Labels: map[string]string{
+						"some-label":  "val",
+						"other-label": "new-val",
+					},
+				},
+			},
+			existingNodes: map[string]map[string]string{
+				"node-1": {
+					"some-label":  "val",
+					"other-label": "val",
+				},
+			},
+			wantNodes: map[string]map[string]string{
+				"node-1": {
+					"some-label":  "val",
+					"other-label": "new-val",
+				},
+			},
+		},
+		{
+			name: "Update node labels with new labels and values for existing node (more than 1 existing nodes)",
+			node: v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "node-2",
+					Labels: map[string]string{
+						"some-label":  "val",
+						"other-label": "new-val",
+						"new-label":   "new-val",
+					},
+				},
+			},
+			existingNodes: map[string]map[string]string{
+				"foo": {
+					"some-label":  "val",
+					"other-label": "val",
+				},
+				"node-2": {
+					"some-label":  "val",
+					"other-label": "val",
+				},
+			},
+			wantNodes: map[string]map[string]string{
+				"foo": {
+					"some-label":  "val",
+					"other-label": "val",
+				},
+				"node-2": {
+					"some-label":  "val",
+					"other-label": "new-val",
+					"new-label":   "new-val",
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			nodeStore := NewNodeStore()
+			if tt.existingNodes != nil && len(tt.existingNodes) > 0 {
+				nodeStore.nodes = tt.existingNodes
+			}
+			nodeStore.SetNode(&tt.node)
+			assert.EqualValues(t, tt.wantNodes, nodeStore.nodes)
+		})
+	}
+}
+
+func Test_GetNodes(t *testing.T) {
+	tests := []struct {
+		name      string
+		wantNodes map[string]map[string]string
+	}{
+		{
+			name:      "Get empty node store",
+			wantNodes: map[string]map[string]string{},
+		},
+		{
+			name: "Get node store with 1 existing node",
+			wantNodes: map[string]map[string]string{
+				"foo": {
+					"some-label":  "val",
+					"other-label": "val",
+				},
+			},
+		},
+		{
+			name: "Get node store with multiple nodes",
+			wantNodes: map[string]map[string]string{
+				"foo": {
+					"some-label":  "val",
+					"other-label": "val",
+				},
+				"bar": {
+					"some-label":  "val",
+					"other-label": "new-val",
+					"new-label":   "new-val",
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			nodeStore := NewNodeStore()
+			if tt.wantNodes != nil && len(tt.wantNodes) > 0 {
+				nodeStore.nodes = tt.wantNodes
+			}
+			gotNodes := nodeStore.GetNodes()
+			assert.EqualValues(t, tt.wantNodes, gotNodes)
+		})
+	}
+}
+
+func Test_UnsetNodes(t *testing.T) {
+	tests := []struct {
+		name          string
+		node          v1.Node
```

Comment:
`UnsetNode` receives a name, not a `v1.Node` so the tests could be simplified a bit if we changed this to be a string instead.